hexrd/instrument.py

```diff
@@ -3573,7 +3580,257 @@ def simulate_laue_pattern(self, crystal_data,
+        """
+        num = uvw.shape[0]
+        cx = np.atleast_2d(self.caxis).T
+        dp = np.dot(uvw, cx)
+        den = np.squeeze(np.sqrt(1 - dp**2))
+        mask = den < 1E-8
+        beta = np.zeros([num, ])
+        beta[~mask] = self.radius/den[~mask]
+        beta[mask] = np.nan
+
+        return np.tile(beta, [3, 1]).T * uvw
+
+    def _clip_to_cylindrical_detector(self, uvw):
+        """
+        takes in the intersection points uvw
+        with the cylindrical detector and 
+        prunes out points which don't actually
+        hit the actual panel
+
+        Parameters
+        ----------
+        uvw : numpy.ndarray
+        unit vectors stacked row wise (nx3) shape
+
+        Returns
+        -------
+        numpy.ndarray
+        (x,y,z) vectors point which fall on panel
+        with (mx3) shape
+        """
+        # first get rid of points which are above
+        # or below the detector
+        size = self.physical_size
+        tvec = np.atleast_2d(self.tvec).T
+        cx = np.atleast_2d(self.caxis).T
+        num = uvw.shape[0]
+        ycomp = uvw - np.tile(self.tvec,[num, 1])
+        ylen = np.squeeze(np.dot(ycomp, cx))
+        mask = np.abs(ylen) <= size[0]*0.5
+        res = uvw[mask, :]
+
+        # next get rid of points that fall outside 
+        # the polar angle range
+        num = res.shape[0]
+        dp = np.squeeze(np.dot(res, cx))
+        v = np.tile(cx,[1, num])*np.tile(dp, [3,1])
+        v = v.T
+        xcomp = res - v
+        magxcomp = np.linalg.norm(xcomp,axis=1)
+        ang = np.squeeze(np.dot(xcomp, tvec))/self.radius/magxcomp
+        ang = np.arccos(ang)
+        mask = ang < self.angle_extent
+        res = res[mask,:]
+
+        return res
+
+    def _gvecToDetectorXY(self, gvecs):
+        """
+        routine to convert gvectors to
+        points on the detector
+        steps as follows:
+        1. gvec to point on cylinder
+        2. prune list to only valid points
+        3. dewarp to rectangle
+        """
+        pass
+
+    @property
+    def caxis(self):
+        # returns the cylinder axis
+        return np.dot(self.rmat, constants.lab_y)
+
+    @property
+    def paxis(self):
+        # returns the cylinder axis
+        return np.dot(self.rmat, constants.lab_x)
+
+    @property
+    def radius(self):
+        # units of mm
+        return np.linalg.norm(self.tvec)
+
+    @property
+    def physical_size(self):
+        # return physical size of detector
+        # in mm after dewarped to rectangle
+        return np.array([self.rows*self.pixel_size_row,
+                         self.cols*self.pixel_size_col])
+
+    @property
+    def beam_position(self):
+        """
+        returns the coordinates of the beam in the cartesian detector
+        frame {Xd, Yd, Zd}.  NaNs if no intersection.
+        """
+        output = np.nan * np.ones(2)
+        pt_on_cylinder = self._unitvec_to_cylinder(
+                         np.atleast_2d(bvec))
```

Comment:
Add indentation here to make it more readable.
```suggestion
                             np.atleast_2d(bvec))
```